### PR TITLE
fastfetch: 2.20.0 -> 2.21.0

### LINKED ISSUE #332426: Update request: fastfetch 2.20.0 → 2.21.0
- Package name: fastfetch
- Latest released version: 2.21.0

- Current version on the unstable channel: 2.20.0
- Current version on the stable/release channel: 2.13.2

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@khaneliman 

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes #332426

[Release notes](https://github.com/fastfetch-cli/fastfetch/releases/tag/2.21.0)

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by donovanglover** [comment]
Since there's a large diff with nixfmt I think we should update this like normal and wait until the treewide format.

Making a separate commit for nixfmt and updating `.git-blame-ignore-revs` is also possible.

**Comment by khaneliman** [comment]
> Since there's a large diff with nixfmt I think we should update this like normal and wait until the treewide format.
> 
> Making a separate commit for nixfmt and updating `.git-blame-ignore-revs` is also possible.

Yes, please split formatting commit from the actual change.

**Comment by khaneliman** [comment]
I see in the release notes mention about libnm -> libdbus. Does anyone use the wifi module to validate? Seems like we should be adding libdbus as a linux dependency.

File: pkgs/by-name/fa/fastfetch/package.nix [hunk 1]
```diff
@@ -65,79 +69,84 @@ stdenv'.mkDerivation (finalAttrs: {
...
+      dbus
+      dconf
+      ddcutil
+      glib
+      hwdata
+      libdrm
+      libpulseaudio
+      libselinux
+      libsepol
+      networkmanager
```
Comment (khaneliman) [inline pkgs/by-name/fa/fastfetch/package.nix:91]: networkmanager -> libdbus per release notes

**Review by khaneliman — CHANGES_REQUESTED** [review]
Split commits for formatting and update dependency per release.

File: pkgs/by-name/fa/fastfetch/package.nix [hunk 2]
```diff
@@ -162,8 +171,12 @@ stdenv'.mkDerivation (finalAttrs: {
   meta = {
     description = "Like neofetch, but much faster because written in C";
     homepage = "https://github.com/fastfetch-cli/fastfetch";
+    changelog = "https://github.com/fastfetch-cli/fastfetch/releases/tag/${finalAttrs.version}";
     license = lib.licenses.mit;
-    maintainers = with lib.maintainers; [ khaneliman ];
+    maintainers = with lib.maintainers; [
+      luftmensch-luftmensch
```
Comment (khaneliman) [inline pkgs/by-name/fa/fastfetch/package.nix:177]: separate commit, as well.

File: pkgs/by-name/fa/fastfetch/package.nix [hunk 3]
```diff
@@ -162,8 +169,12 @@ stdenv'.mkDerivation (finalAttrs: {
...
```
Comment (khaneliman) [inline pkgs/by-name/fa/fastfetch/package.nix:175]: fastfetch: add luftmensch-luftmensch maintainer

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 332464` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>fastfetch</li>
    <li>fastfetch.man</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 332464` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>fastfetch</li>
    <li>fastfetch.man</li>
  </ul>
</details>

**Review by khaneliman — APPROVED** [review]
LGTM

File: pkgs/by-name/fa/fastfetch/package.nix [hunk 4]
```diff
@@ -162,8 +171,12 @@ stdenv'.mkDerivation (finalAttrs: {
   meta = {
     description = "Like neofetch, but much faster because written in C";
     homepage = "https://github.com/fastfetch-cli/fastfetch";
+    changelog = "https://github.com/fastfetch-cli/fastfetch/releases/tag/${finalAttrs.version}";
```
Comment (donovanglover) [inline pkgs/by-name/fa/fastfetch/package.nix:172]: This change should be in a separate commit

File: pkgs/by-name/fa/fastfetch/package.nix [hunk 5]
```diff
@@ -162,8 +171,12 @@ stdenv'.mkDerivation (finalAttrs: {
...
     license = lib.licenses.mit;
-    maintainers = with lib.maintainers; [ khaneliman ];
+    maintainers = with lib.maintainers; [
+      luftmensch-luftmensch
+      khaneliman
+    ];
```
Comment (donovanglover) [inline pkgs/by-name/fa/fastfetch/package.nix:177]: This change shouldn't be in the nixfmt commit

Comment (khaneliman) [reply pkgs/by-name/fa/fastfetch/package.nix:177]: was already addressed

Comment (khaneliman) [reply pkgs/by-name/fa/fastfetch/package.nix:177]: sorry, thought it got removed when the new commit was added

**Review by drupol — CHANGES_REQUESTED** [review]
Is there any added value of having two commits with the same commit log message? Can you please avoid twice the same commits and squash them together?

**Comment by luftmensch-luftmensch** [comment]
@drupol Done

**Review by drupol — APPROVED** [review]

**Review by khaneliman — APPROVED** [review]

File: pkgs/by-name/fa/fastfetch/package.nix (hunk 4 above)
Comment (donovanglover) [reply pkgs/by-name/fa/fastfetch/package.nix:172]: Still in the nixfmt commit but not a serious issue so resolving

Comment (drupol) [reply pkgs/by-name/fa/fastfetch/package.nix:172]: yeah it's not really a big deal.

**Review by donovanglover — APPROVED** [review]

**Review by khaneliman — APPROVED** [review]
